src/decorators/walletOpStack.ts

```diff
@@ -17,19 +25,9 @@ export type WalletOpStackActions<
     // TODO name these params
     args: Parameters<typeof bridgeWriteContract>[1],
   ) => Promise<string>
-  writeUnsafeDepositTransaction: <
-    TAbi extends Abi | readonly unknown[] = typeof optimismPortalABI,
-    TFunctionName extends string = 'depositTransaction',
-    TChainOverride extends Chain | undefined = Chain | undefined,
-  >(
-    args: WriteUnsafeDepositTransactionParameters<
-      TAbi,
-      TFunctionName,
-      TChain,
-      TAccount,
-      TChainOverride
-    >,
-  ) => Promise<WriteContractReturnType>
+  writeUnsafeDepositTransaction: DecoratedAction<WriteUnsafeDepositTransaction>
```

Comment:
Not sure if this abstraction will work since we need to pass the `TChain` from the decorator generics through to the action.

Reply:
Basically, there are mismatches between the decorator action generics vs. stand-alone action generics